src/efi.rs

```diff
@@ -22,27 +23,64 @@ use crate::ostreeutil;
 use crate::util;
 use crate::util::CommandRunExt;
 
-/// The path to the ESP mount
-pub(crate) const MOUNT_PATH: &str = "boot/efi";
+/// The ESP partition label
+pub(crate) const ESP_PART_LABEL: &str = "EFI-SYSTEM";
+
+#[macro_use]
+lazy_static! {
+    /// The path to a temporary ESP mount
+    static ref MOUNT_PATH: PathBuf = {
+        // Create new directory in /tmp with randomly generated name at runtime for ESP mount path.
+        tempfile::tempdir_in("/tmp").expect("Failed to create temp dir for EFI mount").into_path()
+    };
+}
 
 #[derive(Default)]
 pub(crate) struct EFI {}
 
 impl EFI {
     fn esp_path(&self) -> PathBuf {
-        Path::new(MOUNT_PATH).join("EFI")
+        Path::new(&*MOUNT_PATH).join("EFI")
+    }
+
+    fn esp_device(&self) -> PathBuf {
+        Path::new("/dev/disk/by-partlabel/").join(ESP_PART_LABEL)
     }
 
     fn open_esp_optional(&self) -> Result<Option<openat::Dir>> {
+        self.ensure_mounted_esp()?;
         let sysroot = openat::Dir::open("/")?;
         let esp = sysroot.sub_dir_optional(&self.esp_path())?;
         Ok(esp)
     }
     fn open_esp(&self) -> Result<openat::Dir> {
+        self.ensure_mounted_esp()?;
         let sysroot = openat::Dir::open("/")?;
         let esp = sysroot.sub_dir(&self.esp_path())?;
         Ok(esp)
     }
+
+    fn ensure_mounted_esp(&self) -> Result<()> {
+        let mount_point = &Path::new("/").join(&*MOUNT_PATH);
+        let output = std::process::Command::new("mountpoint")
```

Comment:
This isn't bad, but now that we know *our* process is the only one doing the mount (in our mount namespace), I think it'd be better to have a something like `Arc<Mutex<Option<PathBuf>>>` or...actually there's [a lazy_init crate for that](https://docs.rs/lazy-init/0.4.0/lazy_init/) even.

Basically do the tempdir creation + mount as a single locked unit instead.